### PR TITLE
Cookie based authentication in addition to current available ones #392

### PR DESCRIPTION
**Pull request by juangermano-okta**
Proposed implementation for cookie based authentication as described in #392

### CODE REVIEW

**Comment by DanielOverdevest** [comment]
How can I help to review this and bring this in the pycontrib:develop github?

File: tests/tests.py [hunk 1]
```diff
@@ -184,6 +184,7 @@ def __init__(self):
                 else:
                     self.CI_JIRA_USER_PASSWORD = 'sd4s3dgec5fhg4tfsds3434'
 
+                self.CI_JIRA_ADMIN_EMAIL_DOMAINE_NAME = os.environ.get('CI_JIRA_ADMIN_EMAIL_DOMAINE_NAME', 'ssbarnea.33mail.com')
```
Comment (hdost) [inline tests/tests.py]: Is this related to the cookie fix?

File: examples/cookie_auth.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+props = jira.application_properties()
+
+# Find all issues reported by the admin
+issues = jira.search_issues('assignee=admin')
+
+# Find the top three projects containing issues reported by admin
+top_three = Counter(
+    [issue.fields.project.key for issue in issues]).most_common(3)
+
+# import time; time.sleep(65)  # Fake cookie expiration
```
Comment (hdost) [inline examples/cookie_auth.py]: Please remove the commented code.

**Comment by hdost** [comment]
@juangermano-okta can you rebase off dev or segment those changes to things not pertinent to the cookie auth into separate PRs?

**Comment by juangermano-okta** [comment]
@hdost Sure thing.

**Comment by juangermano-okta** [comment]
There's a bug in the implementation: it logins OK in response of a 401, and re-authenticates in case the cookie has expired, but it isn't handling correctly getting a 401 when accessing a resource the user is not _authorized_ to access. The response (401) is interpreted as cookie timeout, so login is attempted again. I'll need to come up with something to handle the case.

**Comment by hdost** [comment]
@juangermano-okta It looks like when a resource is not authorized JIRA spits back https://en.wikipedia.org/wiki/HTTP_403 not a 401, but this is server version 7.2.7

**Comment by juangermano-okta** [comment]
@hdost That's interesting, I'll check it out. I noticed the issue when using the cloud hosted version, but I've got access to a JIRA v7.3.6.

**Comment by juangermano-okta** [comment]
@hdost I wasn't able to reproduce the issue with either JIRA v7.3.6 on-prem or hosted in cloud. As you pointed out, I've been receiving 403 for unauthorized resources, which is correctly handled. Since you opened #484 and resolved pending conflicts, we can continue work there.

**Comment by hdost** [comment]
Sounds like a plan
